packages/flutter/lib/src/painting/shader_warm_up.dart

```diff
@@ -182,29 +182,7 @@ class DefaultShaderWarmUp extends ShaderWarmUp {
       ..layout(const ui.ParagraphConstraints(width: 60.0));
     canvas.drawParagraph(paragraph, const ui.Offset(20.0, 20.0));
 
-    // Construct an image for drawImage related operations
-    const int imageWidth = 10;
-    const int imageHeight = 10;
-    final Uint8List pixels = Uint8List.fromList(List<int>.generate(
-      imageWidth * imageHeight * 4,
-          (int i) => i % 4 < 2 ? 0x00 : 0xFF,  // opaque blue
-    ));
-
-    final Completer<void> completer = Completer<void>();
-    ui.decodeImageFromPixels(pixels, imageWidth, imageHeight, ui.PixelFormat.rgba8888, (ui.Image image) {
-      // Warm up image shaders
-      canvas.translate(0.0, 80.0);
-      canvas.save();
-      final ui.Rect srcRect = ui.Rect.fromLTWH(0.0, 0.0, image.width.toDouble(), image.height.toDouble());
-      canvas.drawImage(image, const ui.Offset(20.0, 20.0), ui.Paint());
-      canvas.translate(80.0, 0.0);
-      canvas.drawImageRect(image, srcRect, ui.Rect.fromLTWH(20.0, 20.0, imageWidth * 0.6, imageWidth * 0.6), paints[0]);
-      canvas.translate(80.0, 0.0);
-      canvas.drawImageRect(image, srcRect, ui.Rect.fromLTWH(10.0, 10.0, imageWidth * 1.5, imageWidth * 1.5), paints[0]);
-      canvas.restore();
-      completer.complete();
-    });
-
-    return completer.future;
+    // No asynchronous work in the default shader warm-up yet.
```

Comment:
Is this work that clients would have to repeat?  If so, would it make sense to just guard this with a bool that defaults to false?

Reply:
Ah I see.  It would probably make more sense to just provide another subclass that extends this one and adds the ImageShader stuff. Of course that doesn't need to be done for this PR to land, more asking for curiosity

Reply:
(You now have an unused import for typed_data though)

Reply:
Yes, I'm currently creating a custom ShaderWarmUp for our internal client for the image shaders. I'll make sure that our internal client is Ok with this before landing it :)